### PR TITLE
Created tests for Board. Fixed getPlayerCell() and getPlayers() implementation

### PR DESCRIPTION
**Pull request by susannah-c**
Created tests for Board. Fixed getPlayerCell() and getPlayers() method in Board.

### CODE REVIEW

File: src/main/java/Entities/Board.java [hunk 1]
```diff
@@ -39,6 +48,7 @@ public List<Player> getPlayers(){
     public void removePlayer(Player player){
         List<Player> newPlayers = players;
         newPlayers.remove(player);
+        this.players = newPlayers;
```
Comment (susannah-c) [inline src/main/java/Entities/Board.java:51]: you can do this.players.remove(player) instead.

**Review by harshkpatel — APPROVED** [review]
All the tests look good coverage wise, if I can think of any edge cases later, I will bring them up to add in future PRs.

File: src/main/java/Entities/Board.java [hunk 2]
```diff
@@ -37,8 +46,7 @@ public List<Player> getPlayers(){
      * @param player The player that must be removed.
      */
     public void removePlayer(Player player){
-        List<Player> newPlayers = players;
-        newPlayers.remove(player);
```
Comment (derrcodes) [inline src/main/java/Entities/Board.java:41]: Good catch on this. The previous version I think had the idea of being correct but it forgot to pass newPlayers into this.players. But either way this implementation is a lot better as doing it the previous way may be a waste of resources due to the introduction of newPlayers.

File: src/main/java/Entities/Board.java [hunk 3]
```diff
@@ -63,7 +71,8 @@ public void removePlayer(Player player){
      * @return This returns a Cell which the Player instance is on.
      */
     public Cell getPlayerCell(Player player){
-        return this.cells.get(this.playerPositions.get(player));
+        int position = player.getPosition();
+        return this.getCell(position);
```
Comment (derrcodes) [inline src/main/java/Entities/Board.java:75]: Great change. I don't think it changes anything in terms of functionality but its a lot more readable.

File: src/test/java/Entities/BoardTest.java [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+import java.util.*;
+
+public class BoardTest {
+    @Test
+    public void testGetPlayers(){
+        Player playerOne = new Player("PlayerOne");
```
Comment (Leuxll) [inline src/test/java/Entities/BoardTest.java:11]: I think that one thing we can all do after is to refactor our test code so that we use the @before annotation to specify the setting up for the tests so that we don't need to have redundant code in every test case. But other than that this looks good.

**Review by derrcodes — APPROVED** [review]
Good changes in terms of fixing bugs (the removePlayer method) and some readability changes.  Good to see that there was also the addition of JavaDoc to the second constructor of the Board class. The tests seem good, they also passed. I don't think we need to make changes to the Board class anymore unless there are conflicting changes related to other classes.

**Review by Leuxll — APPROVED** [review]
See my other comment, but other than that this should be fine for now.
